File: src/Ejercicio4/Ejercicio4.java
package Ejercicio4;

import java.util.Scanner;
import java.util.random.RandomGenerator;

public class Ejercicio4 {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int num = numAlet();

        while(true){
            System.out.print("Ingrese un número: ");
            int n = Integer.parseInt(sc.nextLine());

            if (num == n) {
                System.out.println("HAS ACERTADO!!!");
                        break;
            }
            else if (num < n) System.out.println("El número que estas buscando es mas pequeño....");
            else System.out.println("El número que estas buscando es mayor....");
            System.out.println();
        }

    }

    public static int numAlet(){
        return RandomGenerator.getDefault().nextInt(1,100);
    }
}

File: src/Ejercicio4/Ejercicio4.kt
package Ejercicio4

import kotlin.random.Random

fun main(){

    try {

        val numAlet = Random.nextInt(1, 100)

        while (true){
            println("Ingrese un número: ")
            val n = readln().toInt()

            if (numAlet == n) {
                println("HAS ACERTADO!!!")
                break
            }
            else if (numAlet > n) println("El numero que estas buscando es mayor....")
            else println("El número que estas buscando es menor....")
            println()
        }
    }
    catch (e: Exception){
        println("El valor ingresado no era un número.")
    }

}